### PR TITLE
React library update

### PR DESCRIPTION
**Pull request by spencer-nelson**
Process the components through svgr and tidy up the export with props you can pass to style the fill color.

### CODE REVIEW

**Comment by spencer-nelson** [comment]
The resulting components will look like:

import * as React from "react";
interface IAccessTime24Regular {
  primary?: string;
}

const AccessTime24Regular = (props: IAccessTime24Regular) => {
  return <svg width={24} height={24} fill="none" xmlns="http://www.w3.org/2000/svg"><path d="<snip/>" fill={props.primary} /><path d="<snip>" fill={props.primary} /></svg>;
};

export { IAccessTime24Regular, AccessTime24Regular };

File: packages/react-icons/template.js
Comment (zannstpierre) [inline packages/react-icons/template.js:18]: You might want to expand this to accept more interface properties.

Comment (spencer-nelson) [reply packages/react-icons/template.js:18]: Which properties could/should be needed and why? The only possible ones I can think of would be width/height but those are inherent in the selected icon.

File: packages/react-icons/template.js
Comment (spencer-nelson) [inline packages/react-icons/template.js:17]: Rather than a different interface for each component, would it be better to have one single IFluentIconProps interface that is just used by all of them? That seems more inline with the idea that all the icons are part of a cohesive system.

Comment (zannstpierre) [reply packages/react-icons/template.js:17]: If they could use all use it, that would be ideal.

File: packages/react-icons/package.json
Comment (Philip-Scott) [inline packages/react-icons/package.json:5]: Do you need this author tag? I see that it's been removed on a separate PR here: https://github.com/microsoft/fluentui-system-icons/pull/177

File: packages/react-icons/template.js
Comment (Philip-Scott) [reply packages/react-icons/template.js:17]: I agree with this comment. If all the props are the same for all icons, a single shared IFluentIconProps interface would be ideal and allow for a simpler type usage downstream. Then later on if an icon has it's own props, you could have a union type, kind of like `IFluentIconProps | ISettingsIconProps`

File: packages/react-icons/template.js
Comment (Philip-Scott) [reply packages/react-icons/template.js:18]: Some icons can be filled iirc, is this controlled by a property on the SVG, or is it it's own separate icon?

File: packages/react-icons/package.json
Comment (Philip-Scott) [inline packages/react-icons/package.json:15]: Oh interesting, you are using SVGR. Just as an FYI, we had issues with this tool on a previous usage on the Fluid site due to groups on the SVG itself, but they were more complex shapes than icons. 

Do you have a way to view every icon once it's generated just to make sure svgr didn't cause any artifacts?

File: packages/react-icons/package.json
Comment (Philip-Scott) [inline packages/react-icons/package.json:26]: Is `shx` as a devDependency needed? It appears that it's only used as a wrapper on the clean commands and may be we can get rid of it. I'm not sure what terminal environments you run this so it could be fine, and if so please ignore this comment.

**Review by Philip-Scott — COMMENTED** [review]
One question for clarification, are all icons are 24x24, with the 2px internal padding making the internal icons actually look more as 20x20, correct?

File: packages/react-icons/template.js
Comment (bheston) [reply packages/react-icons/template.js:18]: This is different than the model we discussed earlier, and while it may not be _too_ rigid, I wonder if it's not flexible enough. The interface we've been using that works with the flexibility of css is simply to apply an optional class name. Then any arbitrary style can be applied if necessary. The first reasonable example that comes to mind is something like `margin`, though that could potentially be applied generically to the `svg` element.

If we want to go this route, I think defaulting that value to `currentcolor` would be useful because much of the time, without knowing otherwise, an icon is the same color as neighboring text.

If this is the only way to provide the primary color, or eventually secondary or fill, etc. I wonder if this will be excess work to replicate on many icons. That is the values would always need to be known and passed to each icon.

Also, I'm not sure if this work with css-in-js where the colors might be determined at runtime.

This may be a good option for some consuming patters. I suggest maintaining className support as well.

File: packages/react-icons/package.json
Comment (bheston) [inline packages/react-icons/package.json:14]: I didn't get a chance to experiment with this, but I was thinking the workflow should be to consume the optimized icons from the svg package. That way all logic about optimizing goes there and we can make any future updates in one place.

File: packages/react-icons/package.json
Comment (bheston) [inline packages/react-icons/package.json:2]: @fluentui/react-icons

File: packages/react-icons/package.json
Comment (bheston) [reply packages/react-icons/package.json:26]: I used shx in the svg icons package to allow it to run on Windows. The official build is on Linux, but it was more convenient during dev time I thought.

File: packages/react-icons/package.json
Comment (chrisdholt) [reply packages/react-icons/package.json:2]: We should probably get an owner from Fluent UI React to ensure that the namespace works with their current expectations/requirements (if we haven't already).

File: packages/react-icons/template.js
Comment (chrisdholt) [reply packages/react-icons/template.js:18]: I think clarity on the prop contract here would be helpful. Is there a method to modify the HTML attributes on the SVG itself if needed? Consider a scenario where we need to apply ARIA, add a title attribute for some reason, etc...absolutely agree on className support as well. Are these intended to work only with the Fluent UI react implementation or with others *using* React? If the latter, I think we'll want to work through the prop contract to support a broader implementation.

File: packages/react-icons/package.json
Comment (spencer-nelson) [reply packages/react-icons/package.json:13]: Doing that will require that the SVG package be given any treatments needed for React. Are there any possible uses or properties that the React package will need that would be blocked by the optimizations for the SVG package? That's not my expertise, so I'll defer to others with more experience, but my first concern is the styling options, particularly as color is concerned. We have been asked to address possible multi-color icon scenarios. The way I see for that is to assign a color to individual paths, but all fills have been removed from the SVGs at the moment.

File: packages/react-icons/package.json
Comment (spencer-nelson) [reply packages/react-icons/package.json:25]: On that note, @bheston can you confirm which of these commands might not reasonably work for someone trying to maintain this on a Windows machine? I've always started my Windows dev boxes by installing gnu utils so don't recall which tools like sed/rm/whatever aren't readily available.

File: packages/react-icons/package.json
Comment (tomi-msft) [reply packages/react-icons/package.json]: Yeah, `@fluentui/react-icons` is the expected namespace for this package

File: packages/react-icons/template.js
Comment (spencer-nelson) [reply packages/react-icons/template.js:18]: They are expected to work for any React package. Assuming we define a package interface akin to:
```
export interface IFluentIconsProps {
	primary?: string;
	className?: string;
}
```
does this allow further extension to support other HTML attributes, either by product consumers or as a wider package implementation in the future?

Comment (spencer-nelson) [reply packages/react-icons/template.js:18]: Further expounding on the className, is the following worth pushing as a V1 option, with the interface defined above?
```
import * as React from "react";
import { JSX } from "react-jsx";
import { IFluentIconsProps } from "./IFluentIconsProps.types";

const AccessTime24Regular = (
  iconProps: IFluentIconsProps,
  props: React.HTMLAttributes<HTMLElement>
): JSX.Element => {
  const { primaryFill = "#212121", className } = iconProps;
  return (
    <svg
      width={24}
      height={24}
     
      xmlns="http://www.w3.org/2000/svg"
      className={className}
      {...props}
    >
      <path
        d="<snip/>"
        fill={primaryFill}
      />
      <path
        d="<snip/>"
        fill={primaryFill}
      />
    </svg>
  );
};

export default AccessTime24Regular;
```
It was earlier mentioned to me that we might not want to spread all the props in this way, but I don't know another way to get the HTML attributes in.

File: packages/react-icons/package.json
Comment (spencer-nelson) [reply packages/react-icons/package.json]: Thanks. We don't have a visual regression test set up at the moment to test for that, but given the transforms happening and the simple svgs we build in the first place, we won't have any groups. If those were the breaking elements for you I feel we'll be good to go with svgr, but what did you all settle on using and how difficult would it be to swap in for svgr here?

Comment (Philip-Scott) [reply packages/react-icons/package.json]: I ended up opening the SVG in Inkscape, and ungrouping everything to fix the issue and that worked. Also validated that if our designers gave us SVGs without being grouped, there were no artifacts. This was late 2019 so maybe the issue was fixed, but I thought it was worth mentioning as an FYI :)

**Comment by spencer-nelson** [comment]
> One question for clarification, are all icons are 24x24, with the 2px internal padding making the internal icons actually look more as 20x20, correct?

That is correct. They will visually appear as 20x20 and should be placed 2px offset up and to the left to get them pixel-perfect placement compared to systems without the 2px internal affordance.

File: packages/react-icons/package.json
Comment (Philip-Scott) [reply packages/react-icons/package.json:25]: Due to the open source nature of the icons package, it sounds fine to use to allow someone without the gnu utils to clone and use fully the repo. Thanks for the explanation Brian. 

In that case you are using `rm` and `mv` on the `updatePrefix` and `updateExport` scripts without shx, so those commands may not work on Windows :)

File: packages/react-icons/package.json
Comment (bheston) [reply packages/react-icons/package.json:13]: It seems we'd need/want to support the same color capabilities across all published formats, so I don't see any reason React would need something that we were removing in svg. If anything it allows us to standardize the way we treat the icons, for instance anywhere we support a secondary color or fill it would work the same way at the actual svg level. The interface might vary based on implementation.

File: packages/react-icons/template.js
Comment (bheston) [inline packages/react-icons/template.js:19]: I think the fill needs to default to null for this to work correctly. Many usages won't want to specify a color for each icon at this level. It doesn't account for the separation of the icon as content and the styling as css. We can author the icons with 212121 but there isn't really a default that seems convenient here aside from "currentcolor", which might be undesirable for other uses. The upside here is that null fill everywhere renders as black by default.

File: packages/react-icons/package.json
Comment (spencer-nelson) [reply packages/react-icons/package.json]: In digging around and playing with it, I can't find a way to make it work well on both systems. We either hit issues with wildcard expansion or the variations on looping through the files break on one system or the other. Given my lack of time I'm going to suggest we instead tell Windows devs that want to copy/maintain the pipeline use WSL and we will try to just stick to a Ubuntu-friendly solution as that's what the GitHub workflows use to run them anyway.

**Review by tomi-msft — APPROVED** [review]

File: packages/react-icons/package.json
Comment (spencer-nelson) [reply packages/react-icons/package.json:13]: Given that we don't want to block this, I'll add an issue to handle multi-color options and specifically mention tightening up on this flow.

**Pull request merged**
